clientv3/watch.go

```diff
@@ -612,6 +601,20 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 				// shutdown from closeSubstream
 				return
 			}
+
+			if wr.Created {
+				select {
+				case ws.initReq.retc <- ws.outc:
+				default:
+				}
+
+				// send creation event and only if requested
+				if ws.initReq.createdNotify {
+					ws.outc <- *wr
+				}
+				continue
```

Comment:
This `continue` breaks resume; the create notify binds a revision to the watcher so `ws.initReq.rev` needs to be updated. Otherwise, the watcher for rev=0 can send a create notify at rev=10, disconnect, then resume to the newest revision rev=1000 (since `ws.initReq.rev` remained 0),  skipping some events.